app/models/concerns/account_interactions.rb

```diff
@@ -66,8 +74,15 @@ def follow_mapping(query, field)
     has_many :domain_blocks, class_name: 'AccountDomainBlock', dependent: :destroy
   end
 
-  def follow!(other_account)
-    active_relationships.find_or_create_by!(target_account: other_account)
+  def follow!(other_account, reblogs: nil)
+    reblogs = true if reblogs.nil?
+    rel = active_relationships.create_with(show_reblogs: reblogs).find_or_create_by!(target_account: other_account)
+    if rel.show_reblogs != reblogs
```

Comment:
wouldn't `rel.update!(show_reblogs: reblogs)` be sufficient? rails won't run a query if nothing's changed, right?

Reply:
Yeah, looks like that would be simpler. Thanks!